src/visitors/eval.hpp

```diff
@@ -337,11 +337,16 @@ namespace wpp {
 				// Check if parameter.
 				if (args) {
 					if (auto it = (*args).find(caller_name); it != (*args).end()) {
-						if (caller_args.size() > 0) {
+						if (caller_args.size() > 0)
 							throw wpp::Exception{caller_pos, "calling '", caller_name, "' as if it were a function, it is an argument."};
-						}
 
 						str = it->second;
+
+						// Check if it's shadowing a function (even this one).
+						auto func = functions.find(it->first + "0");
+						if (func != functions.end())
+							wpp::warn(caller_pos, "parameter ", it->first, " is shadowing function ", func->first);
```

Comment:
Use parameter name in place of function name because function is always going to be `name` + `0`. 
```
wpp::warn(caller_pos, "parameter ", it->first, " is shadowing a function");
```